internal/pkg/pct/build.go

```diff
@@ -57,3 +67,36 @@ func (b *Builder) Build(templatePath, targetDir string) (gzipArchiveFilePath str
 
 	return gzipArchiveFilePath, nil
 }
+
+func (b *Builder) checkConfig(configFile string) error {
+
+	fileBytes, err := b.AFS.ReadFile(configFile)
+	if err != nil {
+		return err
+	}
+
+	var info PuppetContentTemplateInfo
+	err = yaml.Unmarshal(fileBytes, &info)
+	if err != nil {
+		return err
+	}
+
+	msg := "The following attributes are missing in pct-config.yml:\n"
+	orig := msg
+	// These parts are essential for build and deployment.
+
+	if info.Template.Id == "" {
```

Comment:
Color me surprised to learn that go automagically handles nil vs empty, that's neat